src/app/admin/user-panel/user-panel.component.ts

```diff
@@ -34,10 +54,42 @@ export class AdminPanelComponent implements OnInit {
   private webApiService = inject(WebApiService);
   authService = inject(AuthService);
 
+  // User management table data
   tableUsersList = new MatTableDataSource<User>([]);
   displayedColumns = ['email', 'roles', 'actions'];
   selectedUser: User | undefined = undefined;
 
+  // Tab state
+  selectedTabIndex = 0;
+
+  // Displayed columns for the user logs
+  userLogsDisplayedColumns = ['email', 'action', 'time'];
+
+  // Pagination state
+  private pageSubject = new BehaviorSubject({
+    page: 1,
+    limit: 50,
+  });
+
+  userLogsLoading = false;
```

Comment:
I think these need to be signals in modern Angular. Though if it is updating as you want, then maybe not? I was trying to review if they entire template/view gets re-rendered on changes or not, but I'm not sure if Angular does sub-view updates. Was skimming these articles, but they're a bit low-level
https://angular.love/the-latest-in-angular-change-detection-zoneless-signals
https://angular.love/signals-in-angular-deep-dive-for-busy-developers